### ISSUE TITLE
heatMap should implement coordinateGridChart

### ISSUE THREAD

**Issue by eileenprice** [dc-js/dc.js#1461]
As pointed out in #1208 heatMap has no support for any axis customization aside from `keyAccessor`, `valueAccessor`, `rowsLabel`, and `colsLabel`. If it implemented coordinateGridChart, you'd be able to add axis labels, customize the number of ticks on the axes, format the tick labels easily using scale domain, and more.
Personally, I am probably just going to add a bunch of the functionality from `coordinateGridMixin.js` to my chart's `pretransition`, but heatMap is based on a coordinate grid enough so that it would make sense to integrate the two.
Is there a reason heatMap doesn't do this, or has it just not been gotten around to yet? (apologies if there's a good reason for this, please help me understand!)

**Comment by gordonwoodhull** [comment]
I agree - I don't see any reason why it's not derived from coordinateGridChart.

dc.js has had a lot of different contributors over the years, and probably the heatmap author didn't understand the hierarchy or couldn't figure out how to integrate with it. 

It's similar with the rowChart, which sort of should be a coordinateGridChart and is missing functionality because it is not. There it's a bit more complicated because it's X against Y instead of Y against X.

The heatmap is very similar to the scatterPlot, which does manage to be a coordinateGridChart (although it does strange things to make it fit, #702).

**Comment by eileenprice** [comment]
Yep, I ran into the same problem with the rowChart as well. I got the impression that it wasn't used very much, though, so I didn't think it was particularly worth raising an issue for. 
Maybe if I get any free time I'll try integrating heatMap and coordinateGridChart :)

**Comment by gordonwoodhull** [comment]
The row chart definitely gets a lot of use. It has the simple, convenient feature that it can fit longer labels, and it's not too hard to add scrolling.

I think a lot of users have become habituated to the inconsistent interfaces and don't give it much thought. 

If you have ideas how to upgrade charts by bringing them better into the hierarchy, that would be a valuable contribution. Since everything has tests, we could be fairly sure of backward compatibility.
